ocs_ci/ocs/bucket_utils.py

```diff
@@ -1307,3 +1307,73 @@ def get_bucket_available_size(mcg_obj, bucket_name):
     resp = bucket_read_api(mcg_obj, bucket_name)
     bucket_size = resp["storage"]["values"]["free"]
     return bucket_size
+
+
+def compare_bucket_contents(mcg_obj, first_bucket_name, second_bucket_name):
+    """
+    Verify replication was performed within an acceptable time
+
+    """
+
+    def _comparison_logic():
+        first_bucket_object_set = {
+            obj.key for obj in mcg_obj.s3_list_all_objects_in_bucket(first_bucket_name)
+        }
+        second_bucket_object_set = {
+            obj.key for obj in mcg_obj.s3_list_all_objects_in_bucket(second_bucket_name)
+        }
+        if first_bucket_object_set == second_bucket_object_set:
+            logger.info("Objects in both buckets are identical")
+            return True
+        else:
+            logger.warning(
+                f"""Buckets {first_bucket_name} and {second_bucket_name} do not contain the same objects.
+                    {first_bucket_name} objects:
+                    {first_bucket_object_set}
+                    {second_bucket_name} objects:
+                    {second_bucket_object_set}
+                    """
+            )
+            return False
+
+    try:
+        for comparison_result in TimeoutSampler(600, 30, _comparison_logic):
+            if comparison_result:
+                return True
+    except TimeoutExpiredError:
+        logger.error("The compared buckets do not contain the same set of objects")
+        assert False, "The compared buckets do not contain the same set of objects"
+
+
+def write_test_objects(
+    mcg_obj,
+    awscli_pod,
+    bucket_to_write,
+    original_dir,
+    amount=1,
+    s3_creds=None,
+):
+    """
+    Upload random objects to a bucket
+    """
+    full_object_path = f"s3://{bucket_to_write}"
+    object_list = []
+
+    for i in range(amount):
+        file_name = f"testfile{i}.txt"
+        object_list.append(file_name)
+        awscli_pod.exec_cmd_on_pod(
+            f"dd if=/dev/urandom of={original_dir}/{file_name} bs=1M count=1 status=none"
```

Comment:
we have the same command in ocs_ci.ocs.bucket_utils.setup_base_objects.
Please save it as a constant and call it from all places